### PR TITLE
Optimise QgsCoordinateTransform::transformBoundingBox

### PR DESCRIPTION
**Pull request by nyalldawson**
Using std::vector is preferable here, as we don't need implicit sharing of the container or want the extra overhead associated with it

### CODE REVIEW

File: src/core/proj/qgscoordinatetransform.cpp [hunk 1]
```diff
@@ -634,9 +634,9 @@ QgsRectangle QgsCoordinateTransform::transformBoundingBox( const QgsRectangle &r
 
   // We're interfacing with C-style vectors in the
   // end, so let's do C-style vectors here too.
-  QVector<double> x( nXPoints * nYPoints );
-  QVector<double> y( nXPoints * nYPoints );
-  QVector<double> z( nXPoints * nYPoints );
+  std::vector<double> x( nXPoints * static_cast< std::size_t >( nYPoints ) );
```
Comment (rouault) [inline src/core/proj/qgscoordinatetransform.cpp:637]: the comment about C-style vector above is (was) a bit misleading :-)

**Review by rouault — APPROVED** [review]

File: src/core/proj/qgscoordinatetransform.cpp [hunk 2]
```diff
@@ -634,9 +634,9 @@ QgsRectangle QgsCoordinateTransform::transformBoundingBox( const QgsRectangle &r
 
   // We're interfacing with C-style vectors in the
   // end, so let's do C-style vectors here too.
```
Comment (nyalldawson) [inline src/core/proj/qgscoordinatetransform.cpp:636]:
```suggestion
```

File: src/core/proj/qgscoordinatetransform.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/proj/qgscoordinatetransform.cpp:635]: Ok, killing it :)

**Comment by agiudiceandrea** [comment]
Please see https://github.com/qgis/QGIS/issues/59821.

**Pull request merged**